cli/flags.rs

```diff
@@ -1315,35 +1317,80 @@ pub fn resolve_urls(urls: Vec<String>) -> Vec<String> {
 }
 
 /// Expands "bare port" paths (eg. ":8080") into full paths with hosts. It
-/// expands to such paths into 3 paths with following hosts: `0.0.0.0:port`,
-/// `127.0.0.1:port` and `localhost:port`.
+/// expands to such paths into 5 paths with following hosts: `0.0.0.0:port`,
+/// `127.0.0.1:port`, `[::]:port`, `[::1]:port` and `localhost:port`.
 fn resolve_hosts(paths: Vec<String>) -> Vec<String> {
+  lazy_static! {
+    // https://tools.ietf.org/html/rfc1035
+    static ref DOMAIN_ALLOWED_CHARACTERS: Regex = Regex::new(r"^[-.a-zA-Z0-9]+$").unwrap();
+  }
+
   let mut out: Vec<String> = vec![];
   for host_and_port in paths.iter() {
+    // pure port:                 :1234
+    // ipv4 with port:            1.2.3.4:1234
+    // ipv6 with port:            [1:2:3::4]:1234
+    // pure ipv4:                 1.2.3.4
+    // pure ipv6 with bracket:    [1:2:3::4]
+    // pure ipv6 without bracket: 1:2:3::4
```

Comment:
Is this standard? How would you know which sections are part of the IP and which section is the port?

Reply:
If this is an IPv6 with the port, the IP must put in the brackets;
Otherwise, if the port is not specified, the brackets around the IP is optional.
So, if it looks like an IPv6 address, but no brackets around it, it'll be used as a pure IP, like IPv4 without the port. and this means all port on this IP address will be permitted.

Reply:
Ah got it. Thanks for the explanation 👍